### PR TITLE
Support `using` local declarations in a Simple Program.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Related to #41704.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@cston, @RikkiGibson, @dotnet/roslyn-compiler Please review

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs [hunk 1]
```diff
@@ -6957,6 +6944,12 @@ private bool IsPossibleLocalDeclarationStatement(bool isGlobalScriptLevel)
                 return true;
             }
 
+            return IsPossibleFirstTypedIdentifierInLocaDeclarationStatement(isGlobalScriptLevel);
+        }
+
+        private bool IsPossibleFirstTypedIdentifierInLocaDeclarationStatement(bool isGlobalScriptLevel)
+        {
+
             bool? typedIdentifier = IsPossibleTypedIdentifierStart(this.CurrentToken, this.PeekToken(1), allowThisKeyword: false);
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Parser/LanguageParser.cs]: Extra blank line.

File: src/Compilers/CSharp/Test/Syntax/Parsing/SimpleProgramsParsingTests.cs [hunk 2]
```diff
@@ -1744,5 +1744,503 @@ public void TestSkippedTest()
...
+                    N(SyntaxKind.IdentifierName);
+                    {
+                        N(SyntaxKind.IdentifierToken, "type");
+                    }
+                    N(SyntaxKind.SemicolonToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
```
Comment (cston) [inline src/Compilers/CSharp/Test/Syntax/Parsing/SimpleProgramsParsingTests.cs]: >} [](start = 8, length = 1)

Consider testing `using static type = identifier;`.

**Comment by AlekseyTs** [comment]
@RikkiGibson, @dotnet/roslyn-compiler Please review, need a second sign-off for a fairly small change.

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs [hunk 3]
```diff
@@ -7053,6 +7047,55 @@ private bool IsPossibleLocalDeclarationStatement(bool isGlobalScriptLevel)
...
+            var tk = PeekToken(1).Kind;
+
+            if (tk == SyntaxKind.RefKeyword)
+            {
+                return true;
+            }
+
+            if (IsDeclarationModifier(tk)) // treat `const int x = 2;` as a local variable declaration
+            {
+                if (tk != SyntaxKind.StaticKeyword) // For `static` we still need to make sure we have a typed dentifier after it, because `using static type;` is a valid using directive.
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Parser/LanguageParser.cs]: nit: typed *identifier

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs [hunk 4]
```diff
@@ -7053,6 +7047,55 @@ private bool IsPossibleLocalDeclarationStatement(bool isGlobalScriptLevel)
...
+                {
+                    return true;
+                }
+            }
+            else if ((SyntaxFacts.IsPredefinedType(tk) &&
+                        this.PeekToken(1).Kind != SyntaxKind.DotToken && // e.g. `int.Parse()` is an expression
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Parser/LanguageParser.cs]: should this use 'tk' instead?

Comment (AlekseyTs) [reply src/Compilers/CSharp/Portable/Parser/LanguageParser.cs]: > should this use 'tk' instead?

Good catch, that should peek at the next token instead, i.e. 2.

---
In reply to: [392430189](https://github.com/dotnet/roslyn/pull/42385#discussion_r392430189) [](ancestors = 392430189)

**Pull request merged**